torch/ao/quantization/fx/prepare.py

```diff
@@ -805,20 +877,43 @@ def _maybe_insert_output_observer_for_node(
     is_standalone_module = node.meta["target_dtype_info"].get("_is_standalone_module", False)
 
     output_act_obs_or_fq_ctr = node.meta["target_dtype_info"].get("output_act_obs_or_fq_ctr")
-    target_dtype, is_dynamic = _get_dtype_and_is_dynamic(output_act_obs_or_fq_ctr)
-    should_insert_observer = target_dtype not in _DO_NOT_OBS_DTYPE_LIST + [torch.float]
-    # TODO(future PR): move the following logic to
-    # should_insert_observer_for_output
-    should_insert_observer = should_insert_observer and \
-        target_dtype in [torch.quint8, torch.qint8, torch.qint32, torch.float16] and \
-        not is_dynamic
+    target_dtype, target_is_dynamic = _get_dtype_and_is_dynamic(output_act_obs_or_fq_ctr)
+    # uncomment after we support reuse_input_obs_or_fq properly by having separate
+    # implemntations for this key instead of reusing the input_output_share_observers
+    # code
+    # reuse_input_obs_or_fq = node.meta["target_dtype_info"].get("reuse_input_obs_or_fq", False)
```

Comment:
but what is the difference between observer sharing and `reuse_input_obs_or_fq`

Reply:
they might be the same, I can follow up to see if I can merge them